diags/zfs-properties.t

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+# check that ZFS dataset properties are set as specified
+
+export PATH=/bin:/usr/bin:/sbin:/usr/sbin
+
+declare -r description="Check zfs dataset properties"
+
+# Source functions
+source ${NODEDIAGDIR:-/etc/nodediag.d}/functions-tap || exit 1
+
+diag_handle_args "$@"
+
+datasets=`zfs list -H | awk '{print $1}'`
+[ -z "$datasets" ] &&  diag_plan_skip "no ZFS datasets" >&2
+
+function verify_property {
+	local fname="verify_property"
+	local dataset=$1
+	local propname=$2
+	local expectedval=$3
+
+	if [ -z "$dataset" ]; then
+		echo "BUG: $fname missing argument dataset" >&2
+		exit 1
+	fi
+
+	if [ -z "$propname" ]; then
+		echo "BUG: $fname missing argument propname" >&2
+		exit 1
+	fi
+
+	if [ -z "$expectedval" ]; then
+		echo "BUG: $fname missing argument expectedval" >&2
+		exit 1
+	fi
+
+	propval=`zfs list -H -o ${propname} ${dataset} 2>/dev/null`
+	if [ "${propval}" != ${expectedval} ] ; then
+		diag_fail "dataset ${dataset} property ${propname} is ${propval}, expected ${expectedval}" >&2
+	else
+		diag_ok "dataset ${dataset} property ${propname} is ${propval}" >&2
+	fi
+}
+
+#
+# number of tests
+#
+num_tests=0
+for dataset in ${datasets}
+do
+	for idx in ${!DIAG_ZFS_DATASET_NAME[@]}
+	do
+		dsetglob=${DIAG_ZFS_DATASET_NAME[$idx]}
+
+		if [[ ! ${dataset} =~ ${dsetglob} ]]; then
+			continue
+		fi
+
+		if [ -n "${DIAG_ZFS_RECORDSIZE[$idx]}" ] ; then
+			num_tests=$((num_tests+1))
+		fi
+
+		if [ -n "${DIAG_ZFS_DNODESIZE[$idx]}" ] ; then
+			num_tests=$((num_tests+1))
+		fi
+
+		if [ -n "${DIAG_ZFS_XATTR[$idx]}" ] ; then
+			num_tests=$((num_tests+1))
+		fi
+
+		if [ -n "${DIAG_ZFS_CANMOUNT[$idx]}" ] ; then
+			num_tests=$((num_tests+1))
+		fi
+
+		if [ -n "${DIAG_ZFS_COMPRESSION[$idx]}" ] ; then
+			num_tests=$((num_tests+1))
+		fi
+	done
+done
+diag_plan $num_tests
+
+#
+# Main
+#
+
+for dataset in ${datasets}
+do
+	for idx in ${!DIAG_ZFS_DATASET_NAME[@]}
+	do
+		dsetglob=${DIAG_ZFS_DATASET_NAME[$idx]}
+
+		if [[ ! ${dataset} =~ ${dsetglob} ]]; then
+			continue
+		fi
+
+		if [ -n "${DIAG_ZFS_RECORDSIZE[$idx]}" ] ; then
+			verify_property ${dataset} recordsize ${DIAG_ZFS_RECORDSIZE[$idx]}
+		fi
+
+		if [ -n "${DIAG_ZFS_DNODESIZE[$idx]}" ] ; then
+			verify_property ${dataset} dnodesize ${DIAG_ZFS_DNODESIZE[$idx]}
+		fi
+
+		if [ -n "${DIAG_ZFS_XATTR[$idx]}" ] ; then
+			verify_property ${dataset} xattr ${DIAG_ZFS_XATTR[$idx]}
+		fi
+
+		if [ -n "${DIAG_ZFS_CANMOUNT[$idx]}" ] ; then
+			verify_property ${dataset} canmount ${DIAG_ZFS_CANMOUNT[$idx]}
+		fi
+
+		if [ -n "${DIAG_ZFS_COMPRESSION[$idx]}" ] ; then
+			verify_property ${dataset} compression ${DIAG_ZFS_COMPRESSION[$idx]}
+		fi
+	done
+	echo >&2
```

Comment:
Not sure if you want this echo statement there as it adds a newline under the test count that I don't believe other tests have.

Reply:
Yes, that's a mistake.  Thank you.

Reply:
fixed